inc/ticket.class.php

```diff
@@ -4865,18 +4867,60 @@ function showForm($ID, $options = []) {
-         !$canupdate
-      );
+      Html::initEditorSystem($content_id, $rand, true, !$canupdate);
+      if ($ID > 0) {
+         $content = Html::clean($content, false, 1);
+      }
+
+      // If content does not contain <br> or <p> html tag, use nl2br
+      if (!preg_match("/<br\s?\/?>/", $content) && !preg_match("/<p>/", $content)) {
+         $content = nl2br($content);
+      }
 
       echo "<div id='content$rand_text'>";
       if ($canupdate || $can_requester) {
-         echo "<textarea id='$content_id' name='content' style='width:100%' rows='$rows'".
-               ($tt->isMandatoryField('content') ? " required='required'" : '') . ">" .
-               $content."</textarea></div>";
+         Html::textarea([
+            'name'            => 'content',
+            'editor_id'       => $content_id,
+            'required'        => $tt->isMandatoryField('content'),
+            'rows'            => $rows,
+            'enable_richtext' => true,
+            'value'           => $content,
+         ]);
+         if (isset($this->input['_content'])) {
+            foreach ($this->input['_content'] as $uploadId => $upload) {
+               $prefix  = substr($upload, 0, 23);
+               $display = substr($upload, 23);
+               $extension = pathinfo(GLPI_TMP_DIR . '/' . $upload, PATHINFO_EXTENSION);
+               $extensionIcon = '/pics/icones/' . $extension . '-dist.png';
+               if (!is_readable(GLPI_ROOT . $extensionIcon)) {
+                  $extensionIcon = '/pics/icones/defaut-dist.png';
+               }
+               $extensionIcon = $CFG_GLPI['root_doc'] . $extensionIcon;
+               $upload = [
+                  'name'    => $upload,
+                  'id'      => 'doc' . 'content' . mt_rand(),
+                  'display' => $display,
+                  'size'    => filesize(GLPI_TMP_DIR . '/' . $upload),
+                  'prefix'  => $prefix,
+               ];
+               $tag = $this->input['_tag_content'][$uploadId];
+               $tag = [
+                  'name' => $tag,
+                  'tag'  => "#$tag#",
+               ];
+               echo "<p id='" . $upload['id'] . "'>&nbsp;";
+               echo "<img src='$extensionIcon' title='$extension'>&nbsp;";
+               echo "<b>" . $upload['display'] . "</b>&nbsp;(" . Toolbox::getSize($upload['size']) . ")";
+               echo Html::hidden('_content[' . $uploadId . ']', ['value' => $upload['name']]);
+               echo Html::hidden('_prefix_content[' . $uploadId . ']', ['value' => $upload['prefix']]);
+               echo Html::hidden('_tag_content[' . $uploadId . ']', ['value' => $tag['name']]);
+               $deleteUpload = "deleteImagePasted({0:'" . $upload['id'] . "', 1:'" . $upload['id'] . "'+'2'}, '" . $tag['tag'] . "', tinymce.get('$content_id'))";
+               echo '<span class="fa fa-times-circle pointer" onclick="' . $deleteUpload . '"></span>';
+               echo "</p>";
+            }
+         }
```

Comment:
the above should move to Html::textarea() to properly handle uploads everywhere in GLPI (and plugins). This would require a nex key in the options array with a sub array of 3 keys :  _field, _tag_field and  _content_field